### DISCUSSION TITLE
Smart way to report an error in the callback of uv_poll_start?

### DISCUSSION THREAD

**Discussion by r00ta** [libuv/libuv#4120] [Q&A]
Hi there! 

I faced an [issue](https://github.com/twisted/twisted/issues/11904) in an environment under heavy load (network and CPU overloaded) and I'm struggling to reproduce the same issue in a dev environment. 

In short, `uvloop` uses `uv_poll_start` with a callback like 

```
cdef void __on_uvpoll_event(uv.uv_poll_t* handle,
                            int status, int events) with gil:

    if __ensure_handle_data(<uv.uv_handle_t*>handle, "UVPoll callback") == 0:
        return

    cdef:
        UVPoll poll = <UVPoll> handle.data

    if status < 0:
        exc = convert_error(status)
        poll._fatal_error(exc, False)
        return
```

and I was trying to proof that if an error is reported by `libuv` from `uv_poll_start`, then `uvloop` closes the connection and on twisted we don't get any callback. 

Now, my question is: what are the scenarios where such error is reported and how to reproduce them? 

I know I might patch `libuv` and force it to report such error in some specific cases, but I'm wondering if you have a smart way to do it without patching the library. 

Thanks!

**Comment by bnoordhuis** [comment]
At the moment the only time `uv_poll_start()` returns an error is when you're trying to poll a file a descriptor that's already being polled.

**Reply by r00ta** [reply]
Hey @bnoordhuis thank you very much for the reply! I understand the `uv_poll_start()` itself returns an error in that case, but here I'm more trying to get the error notification in the callback `__on_uvpoll_event`. For example, I wrote an application that keeps establishing connections, the server calls `uv_poll_start` and then the client closes the connection - I get an error in the callback let's say once every 100 connections but I don't have a reproducible way to force it.

**Reply by bnoordhuis** [reply]
UV_EBADF? That  means another thread closed the file descriptor libuv is polling.
